pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -820,6 +823,27 @@ func (og *operationGenerator) GenerateVerifyControllerAttachedVolumeFunc(
 	}, nil
 }
 
+func (og *operationGenerator) GenerateVerifyControllerDetachedVolumeFunc(
+	volumeToDetach AttachedVolume,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) (func() error, error) {
+	return func() error {
+		if !volumeToDetach.PluginIsAttachable {
+			// If the volume does not implement the attacher interface, it is
+			// assumed to be detached and the actual state of the world is
+			// updated accordingly.
+			actualStateOfWorld.MarkVolumeAsDetached(volumeToDetach.VolumeName, volumeToDetach.NodeName)
+			return nil
+		}
+
+		glog.Infof("Controller detached volume %q (spec.Name: %q) devicePath: %q",
+			volumeToDetach.VolumeName,
+			volumeToDetach.VolumeSpec.Name(),
+			volumeToDetach.DevicePath)
+		actualStateOfWorld.MarkVolumeAsDetached(volumeToDetach.VolumeName, volumeToDetach.NodeName)
+		return nil
```

Comment:
Yes, VerifyControllerAttachedVolumeFunc check node status. But we could not change this because device path for some volume plugins such as aws needs to be passed by node status.